pyasdf/tests/test_asdftypes.py

```diff
@@ -180,3 +181,27 @@ def url_mapping(self):
 
     del versioning._version_map['42.0.0']
     versioning.supported_versions.pop()
+
+
+def test_longest_match():
+    class FancyComplexExtension(object):
+        @property
+        def types(self):
+            return []
+
+        @property
+        def tag_mapping(self):
+            return []
+
+        @property
+        def url_mapping(self):
+            return [('http://stsci.edu/schemas/asdf/core/',
+                     'FOOBAR/{url_suffix}')]
+
+    l = extension.AsdfExtensionList(
+        [extension.BuiltinExtension(), FancyComplexExtension()])
+
+    assert l.url_mapping(
+        'http://stsci.edu/schemas/asdf/core/asdf-1.0.0') == 'FOOBAR/asdf-1.0.0'
```

Comment:
Maybe, but I think it exposes a possible loophole we'd probably prefer to close :)